lib/filewatch/read_mode/handlers/read_file.rb

```diff
@@ -5,6 +5,10 @@ class ReadFile < Base
     def handle_specifically(watched_file)
       if open_file(watched_file)
         add_or_update_sincedb_collection(watched_file) unless sincedb_collection.member?(watched_file.sincedb_key)
+        if sincedb_collection.member?(watched_file.sincedb_key)
+          previous_pos = sincedb_collection.find(watched_file).position
+          watched_file.file_seek([watched_file.bytes_read, previous_pos].max)
+        end
         loop do
```

Comment:
Since `add_or_update_sincedb_collection` ensures the file is in the sincedb and sets the correct position from the sincedb to the watched_file, can't we simplify this a lot to just be:

```suggestion
      if open_file(watched_file)
        add_or_update_sincedb_collection(watched_file)
        watched_file.file_seek(watched_file.bytes_read)
        loop do
```

Reply:
@jsvd good point! the semantic remains the same and the flow is simplified

Reply:
Without the guard
```ruby
add_or_update_sincedb_collection(watched_file) unless sincedb_collection.member?
```
it happens that `watched_file.bytes_read` is always updated to the `incedb_collection.find(watched_file).position`.
Now this `position` comes from the sinceDB and it's the last pin point.

What happens in this test? The objective of the test is to verify stripped-reads from a couple of files:
**file1**
```
string1\nstring2
```
**file2**
```
stringA\nstringB
```

Striping means that it reads one line for each one, so the `position` point `\n` character.
The test fails because has a `file_chunk_size` is 10 bytes, so the first run for the first file it reads 10 bytes and put in a buffer, which is: 
```
string1\nst
```
if seek set filepointer to the filepoistion (8th bytes) then the next chuck grabbed from the file is 
```
string2
```

which goes in the buffer which contained `st` creating the string `ststring2`.
This is the reason to have the `max(bytes_read, last_pos)`.